kasa/tapo/tapodevice.py

```diff
@@ -0,0 +1,160 @@
+"""Module for a TAPO device."""
+import base64
+import logging
+from datetime import datetime, timedelta, timezone
+from typing import Any, Dict, Optional, Set, cast
+
+from ..aesprotocol import TPLinkAes
+from ..credentials import Credentials
+from ..exceptions import AuthenticationException
+from ..smartdevice import SmartDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TapoDevice(SmartDevice):
+    """Base class to represent a TAPO device."""
+
+    def __init__(
+        self,
+        host: str,
+        *,
+        port: Optional[int] = None,
+        credentials: Optional[Credentials] = None,
+        timeout: Optional[int] = None,
+    ) -> None:
+        super().__init__(host, port=port, credentials=credentials, timeout=timeout)
+        self._state_information: Dict[str, Any] = {}
+        self._discovery_info: Optional[Dict[str, Any]] = None
+        self.protocol = TPLinkAes(host, credentials=credentials, timeout=timeout)
+
+    async def update(self, update_children: bool = True):
+        """Update the device."""
+        if self.credentials is None or self.credentials.username is None:
+            raise AuthenticationException("Tapo plug requires authentication.")
+
+        self._info = await self.protocol.query("get_device_info")
+        self._usage = await self.protocol.query("get_device_usage")
+        self._time = await self.protocol.query("get_device_time")
+
+        self._last_update = self._data = {
+            "info": self._info,
+            "usage": self._usage,
+            "time": self._time,
+        }
+
+        _LOGGER.debug("Got an update: %s", self._data)
+
+    @property
+    def sys_info(self) -> Dict[str, Any]:
+        """Returns the device info."""
+        return self._info
+
+    @property
+    def model(self) -> str:
+        """Returns the device model."""
+        return str(self._info.get("model"))
+
+    @property
+    def alias(self) -> str:
+        """Returns the device alias or nickname."""
+        return base64.b64decode(str(self._info.get("nickname"))).decode()
+
+    @property
+    def time(self) -> datetime:
+        """Return the time."""
+        return datetime.fromtimestamp(
+            cast(float, self._time.get("timestamp")),
+            tz=timezone(timedelta(minutes=cast(float, self._time.get("time_diff")))),
+        )
+
+    @property
+    def timezone(self) -> Dict:
+        """Return the timezone and time_difference."""
+        return {
+            "timezone": timezone(
+                timedelta(minutes=cast(float, self._time.get("time_diff")))
+            )
```

Comment:
I don't think this is correct, should we rather return a timezone object based on the `region` attribute?

Reply:
The `plugP100.TapoDevice` class was using `ZoneInfo` which is python 3.9 only.  The timezone class needs an offset as I understand it.

Because you said your device was returning a [blank string for region](https://github.com/python-kasa/python-kasa/pull/552#discussion_r1406835288) I have updated the logic here to use the region if present, otherwise it will print UTC+-HOURS:

I think it works correctly because I fixed a ValueError @ezekieldas was getting but it's actually difficult for me to tell because I'm in UTC.  Let me know how it works for you.

Reply:
Looks fine here too, let's get this merged! :+1: